src/meta-srv/src/service/heartbeat.rs

```diff
@@ -111,6 +113,7 @@ impl heartbeat_server::Heartbeat for Metasrv {
                 }
             }
 
+            ctx.on_node_disconnect().await;
```

Comment:
Brief network jitters might accidentally delete important data, so it's best to initiate a background task to regularly clean up expired data.